src/ducks/files/Container.jsx

```diff
@@ -38,6 +41,12 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
         confirm(<DeleteConfirm t={ownProps.t} fileCount={files.length} />)
           .then(() => dispatch(trashFiles(files)))
           .catch(() => {})
+    },
+    singleSelection: {
+      rename: selected => new Promise(resolve => {
+        dispatch(startRenaming(selected[0]))
+        resolve()
+      })
```

Comment:
Why is there a `Promise` here? As we use `redux-thunk` we already have async dispatch. It adds complexity.